persistence/scaladsl/src/main/scala/com/lightbend/lagom/internal/scaladsl/persistence/AbstractPersistentEntityRegistry.scala

```diff
@@ -35,36 +37,16 @@ class AbstractPersistentEntityRegistry(system: ActorSystem) extends PersistentEn
     queryPluginId.map(id => PersistenceQuery(system).readJournalFor[EventsByTagQuery](id))
 
   private val sharding = ClusterSharding(system)
-  private val conf = system.settings.config.getConfig("lagom.persistence")
-  private val snapshotAfter: Option[Int] = conf.getString("snapshot-after") match {
-    case "off" => None
-    case _     => Some(conf.getInt("snapshot-after"))
-  }
-  private val maxNumberOfShards: Int = conf.getInt("max-number-of-shards")
-  private val role: Option[String] = conf.getString("run-entities-on-role") match {
-    case "" => None
-    case r  => Some(r)
-  }
-  private val passivateAfterIdleTimeout: Duration = {
-    val durationMs = conf.getDuration("passivate-after-idle-timeout", TimeUnit.MILLISECONDS)
-    if (durationMs == 0) {
-      // Scaladoc of setReceiveTimeout says "Pass in `Duration.Undefined` to switch off this feature."
-      Duration.Undefined
-    } else {
-      durationMs.millis
-    }
-  }
-
-  private val askTimeout: FiniteDuration = conf.getDuration("ask-timeout", TimeUnit.MILLISECONDS).millis
-  private val shardingSettings = ClusterShardingSettings(system).withRole(role)
+  protected val config = PersistenceConfig(system.settings.config.getConfig("lagom.persistence"))
```

Comment:
Could we push this change forward and inject `PersistenceConfig` instead of building it sneakily from `system.settings.config`?  (happy to do it on a separate PR)

Related to https://github.com/lagom/lagom/issues/1255

Reply:
Happy to do that, it would introduce some binary incompatible changes, but only on internal constructors.